src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs

```diff
@@ -81,7 +81,10 @@ public static void ExtractToFile(this ZipArchiveEntry source, string destination
                 ExtractExternalAttributes(fs, source);
             }
 
-            File.SetLastWriteTime(destinationFileName, source.LastWriteTime.DateTime);
+            if (!OperatingSystem.IsAndroid()) // https://github.com/dotnet/runtime/issues/35374, https://github.com/mono/mono/issues/17133
+            {
+                File.SetLastWriteTime(destinationFileName, source.LastWriteTime.DateTime);
+            }
```

Comment:
Should we instead just try/catch so that we still do this when it's supported?

Reply:
That was my thought too. Also, it would be good to avoid issue links in the product code if we can. Source control is great for finding out why code was changed, and in my experience these links tend to build up and obscure the code.

Reply:
A catch would also allow this to work on versions of Android if any that support this.

Reply:
Initially I wanted to use catch, but I was afraid that reviewers are going to scream at me for doing that (ideally I would pass a flag to `File.SetLastWriteTime` to ignore this error on Android, but this would require much more source code changes)

Reply:
I've changed the code to use catch, PTAL